p2p/src/sync/peer_v2/transaction_manager.rs

```diff
@@ -147,6 +157,12 @@ where
                     self.handle_local_event(event)?;
                 }
 
+                _ = self.pending_transactions.due() => {
+                    if let Some(new_tx) = self.pending_transactions.pop(){
```

Comment:
Btw, you can merge both pop and due together and make something like async_pop where the returned value is something you can use directly, instead of having to pop then check.

Reply:
I originally had it that way, but Lukas made a point that it's harder to prove cancelation safety so a split it.

Reply:
Alrighty